tests/containers/kubectl.pm

```diff
@@ -0,0 +1,69 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test the kubectl utility
+# - List pods
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use utils;
+use version_utils;
+use publiccloud::utils;
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
+use containers::k8s;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    # kubectl is in the container module (on SLES)
+    add_suseconnect_product(get_addon_fullname('contm')) if (is_sle);
+    zypper_call("-n in kubernetes-client");
+
+    # Record kubectl version and check if the tool itself is healthy
+    record_info("kubectl", script_output("kubectl version --client --output=json"));
+
+    # Configure CSP/k3s. Only one CSP at a time is possible due to the conflicting configuration in .kube/config
+    assert_script_run('mkdir -p ~/.kube');
+    my $provider = get_var("KUBECTL_CLUSTER", "k3s");
+    if ($provider eq "AKS") {
+        $provider = $self->provider_factory(service => 'AKS');
+    } elsif ($provider eq "k3s") {
+        install_k3s();
+
+        # Ensure k3s has not installed its own kubectl
+        assert_script_run("! stat /usr/local/bin/kubectl");
+        validate_script_output("whereis kubectl", sub { $_ !~ m/\/usr\/local\/bin/ });
+    } else {
+        die "Invalid KUBECTL_CLUSTER defined";
+    }
+
+    # Function to check if the output either has no resources or has at least two lines
+    # Two lines because one line is the header and the following lines are the data rows
+    validate_script_output("kubectl cluster-info", qr/Kubernetes control plane/);
+    # Check for default namespaces to be present
+    validate_script_output("kubectl get namespaces", qr/default/);
```

Comment:
mm not sure that in all providers (e.g. AKS) this namespace will be created by default

Reply:
Yes, I know, when adding more providers we will need to tweak those checks accordingly. I don't know a better way here than to start with Tumbleweed, and then add provider by provider with the corresponding changes to the test cases. WDYT?

Reply:
Answering the previous question, maybe we could cover K3s now adding the others cloud providers in future PR. in this way creating your own namespace is not going to affect anything. I suggest to create the namespace adding the job id or some similar, to avoid overlapping on tests